src/App.tsx

```diff
@@ -15,37 +18,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  name = 'name',
+  length = 'length',
+}
+
+const SORT_FIELD_NAME: SortField = SortField.name;
+const SORT_FIELD_LENGTH: SortField = SortField.length;
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: string,
+  reverse: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return reverse ? preparedGoods.reverse() : preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortField,
+    reverseField,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SORT_FIELD_NAME,
+          })}
+          onClick={() => setSortField(SORT_FIELD_NAME)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SORT_FIELD_LENGTH,
+          })}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
```

Comment:
[UX] - Make sure to mark the active button to explain to the user what action is active. Currently, the 'Sort by length' button does not clearly indicate when it is active.